www/class/centreonCustomView.class.php

```diff
@@ -719,31 +784,52 @@ public function addPublicViewWidgetParams($viewId, $userId)
     }
 
     /**
-     * @param $params
+     * @param int $customViewId
+     * @param int[] $lockedUsers
+     * @param int[] $unlockedUsers
+     * @param int[] $lockedUsergroups
+     * @param int[] $unlockedUsergroups
+     * @param bool $permission
+     * @param bool $authorized
      * @param $userId
      * @throws Exception
+     * @throws CentreonCustomViewExpection
      */
-    public function shareCustomView($params, $userId)
-    {
+    public function shareCustomView(
+        int $customViewId,
+        array $lockedUsers,
+        array $unlockedUsers,
+        array $lockedUsergroups,
+        array $unlockedUsergroups,
+        int $userId,
+        bool $permission,
+        bool $authorized
+    ) {
+        if (!$authorized || !$permission) {
+            throw new CentreonCustomViewException('You are not allowed to share the view');
+        }
         global $centreon;
         $queryValue = array();
 
-        if ($this->checkPermission($params['custom_view_id'])) {
+        if ($this->checkPermission($customViewId)) {
             //////////////////////
             // share with users //
             //////////////////////
             $sharedUsers = array();
             $alwaysSharedUsers = array();
-            $params['lockedUsers'] = isset($params['lockedUsers']) ? $params['lockedUsers'] : array();
-            foreach ($params['lockedUsers'] as $lockedUser) {
-                if ($lockedUser != $centreon->user->user_id) {
-                    $sharedUsers[$lockedUser] = 1;
+
+            if (!empty($lockedUsers)) {
+                foreach ($lockedUsers as $lockedUser) {
+                    if ($lockedUser != $centreon->user->user_id) {
+                        $sharedUsers[$lockedUser] = 1;
+                    }
                 }
             }
```

Comment:
Don't need to check if array is empty, it will already avoid loop if empty
```suggestion
            foreach ($lockedUsers as $lockedUser) {
                if ($lockedUser != $centreon->user->user_id) {
                    $sharedUsers[$lockedUser] = 1;
                }
            }
```